lib/HttpClient.js

```diff
@@ -407,6 +385,9 @@ function rawRequest(opts, cb) {
                 socket.setKeepAlive(true);
             }
 
+            // eagerly return here with the request. it's still very possible
+            // for the request to timeout at this point. if that happens, a
+            // RequestTimeoutError will be emitted as part of the result event.
             cb(null, req);
```

Comment:
Is there a specific reason why we call `cb` here instead of in the `proto.request` callback? Without being familiar with restify client's implementation, I would think the main difference would be that by calling `cb` here it's called when the connection to the peer is established, but possibly before the first part of the HTTP response was received,  whereas by calling it from `proto.request`'s callback, it'd be called only after the first part of the HTTP response was received.

Is that distinction useful? If not, would that allow us to refactor the code in a way that doesn't require us to still write some error prone code to avoid calling `cb` more than once?